system/jlib/jarray.cpp

```diff
@@ -103,6 +98,26 @@ void Allocator::_reallocate(aindex_t newLen, size32_t itemSize)
 };
 
 
+void Allocator::_reallocateExact(aindex_t newLen, size32_t itemSize)
+{
+    if (max == newLen)
+        return;
+
+    if (used > newLen)
+        throw MakeStringException(0, "Reallocate an array would contain too few items: itemSize = %d, trying to allocate %d items contains %u",itemSize, newLen, used);
+
+    size_t allocSize = (size_t)itemSize * newLen;
+    void *newhead = realloc(_head, allocSize);
+    if (!newhead)
+    {
+        IERRLOG("Out of memory in Array allocator: itemSize = %d, trying to allocate %d items", itemSize, max);
```

Comment:
trivial: should be %u's ?

Reply:
It was copied code.  I will change them all.